src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -152,25 +212,64 @@ Task FlushAsync(bool async)
 
         #region Read
 
-        public override int Read(byte[] buffer, int offset, int count) => Read(buffer, offset, count, false).GetAwaiter().GetResult();
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            ValidateArguments(buffer, offset, count);
+            return Read(new Span<byte>(buffer, offset, count));
+        }
 
         public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
+            ValidateArguments(buffer, offset, count);
             if (cancellationToken.IsCancellationRequested)
                 return Task.FromCanceled<int>(cancellationToken);
             using (NoSynchronizationContextScope.Enter())
-                return Read(buffer, offset, count, true);
+                return ReadAsync(new Memory<byte>(buffer, offset, count), cancellationToken).AsTask();
```

Comment:
The cancellation token check and NoSynchronizationContextScope.Enter are called twice, once here and once inside ReadAsync(Memory). Since the byte[] overload is a pure wrapper, the checks can be left only in the Memory overload.

Reply:
OK.